devel/githubContrib.go

```diff
@@ -0,0 +1,253 @@
+	Check(err, "Unable to read response")
+	succ := resp.StatusCode
+	log.Printf("Got %d : %s for %s", succ, resp.Status, url)
+	if succ != http.StatusOK {
+		os.Exit(1)
+	}
+	return body
+}
+
+// PrettyPrintJSON outputs indented version of the Json body (debug only)
+func PrettyPrintJSON(body []byte) {
+	var out bytes.Buffer
+	err := json.Indent(&out, body, "", "  ")
+	Check(err, "Unable to Indent json")
+	_, err = os.Stdout.Write(out.Bytes())
+	Check(err, "Unable to output json")
+}
+
+// Repo is what we use from github rest api v3 listing repositories per org
+type Repo struct {
+	ID              int64  `json:"id"`
+	Name            string `json:"name"`
+	FullName        string `json:"full_name"`
+	ContributorsURL string `json:"contributors_url"`
+	IsFork          bool   `json:"fork"`
+}
+
+// UserC is what we care about from what we get from the ContributorsURL
+type UserC struct {
+	Login         string `json:"login"`
+	ID            int64  `json:"id"`
+	Contributions int64  `json:"contributions"`
+}
+
+// UserData is for the json we get from the /users/:username API call
+type UserData struct {
+	Name    string `json:"name"` // full name
+	Company string `json:"company"`
+	Email   string `json:"email"`
+}
+
+var fromEmailCount = 0 // global variable ftl (or ftw)
+
+// RemoveFromEnd returns a string without the suffix if found
+func RemoveFromEnd(s string, what string) string {
+	if !strings.HasSuffix(s, what) {
+		return s
+	}
+	return s[:len(s)-len(what)]
+}
+
+// GetCompany returns its best guess of the company for a given GitHub user login
+func GetCompany(login string, contribCount int64) string {
+	url := GitHubAPIURL("users/" + login)
+	body := GetBodyForURL(url)
+	var userData UserData
+	err := json.Unmarshal(body, &userData)
+	Check(err, "Unable to parse json")
+	var company string
+	if userData.Company != "" {
+		company = strings.Replace(userData.Company, " ", "", -1)
+	} else if userData.Email != "" {
+		fromEmailCount++
+		company = strings.Split(userData.Email, "@")[1]
+	}
+	// Many people have an @ and/or trailing spaces:
+	company = strings.ToLower(strings.TrimLeft(strings.TrimSpace(company), "@"))
+	company = RemoveFromEnd(company, ".com")
+	company = RemoveFromEnd(company, ".")
+	company = RemoveFromEnd(company, "inc")
+	company = RemoveFromEnd(company, ",")
+	company = RemoveFromEnd(company, ".")
+	// also treat gmail as unknown
+	if company != "" && company != "gmail" {
+		return strings.ToUpper(company[:1]) + company[1:]
+	}
+	log.Printf("%s (%s) has %d contributions but no company nor email", login, userData.Name, contribCount)
+	return "Unknown"
+}
+
+// --- Main --
+
+const minContributions = 3
+const debugJSON = false
+
+func main() {
+	// fmt.Println("in main")
+
+	// Get the repos for the org:
+	const org = "istio"
+	url := GitHubAPIURL("orgs/" + org + "/repos")
+	body := GetBodyForURL(url)
+	var repos []Repo
+	err := json.Unmarshal(body, &repos)
+	Check(err, "Unable to parse json")
```

Comment:
as elsewhere, i'd prefer:

```
if err := json.Unmarshal(body, &repos); err != nil {
   log.Fatalf("Unable to parse json: %v", err) 
}
```

It would be more idiomatic, and easier to understand.